### PR TITLE
Improved completion filtering by visibility

### PR DESCRIPTION
**Pull request by vlad20012**
1. Filter private fields
2. Filter `#[doc(hidden)]` items

### CODE REVIEW

File: src/main/kotlin/org/rust/lang/core/resolve/Processors.kt [hunk 1]
```diff
@@ -150,8 +150,10 @@ fun processAllWithSubst(
 
 fun filterCompletionVariantsByVisibility(processor: RsResolveProcessor, mod: RsMod): RsResolveProcessor {
     return fun(it: ScopeEntry): Boolean {
-        val visible = it.element as? RsVisible ?: return processor(it)
-        if (visible.isVisibleFrom(mod)) return processor(it)
-        return false
+        val element = it.element
+        if (element is RsVisible && !element.isVisibleFrom(mod)) return false
+        if (element is RsOuterAttributeOwner && element.queryAttributes.isDocHidden) return false
```
Comment (Undin) [inline src/main/kotlin/org/rust/lang/core/resolve/Processors.kt]: Are you sure that we should filter such items if they are from our crate?

Comment (vlad20012) [reply src/main/kotlin/org/rust/lang/core/resolve/Processors.kt]: Umm, yep, I remembered that in the case of macro we also check that it's in another module. I should add the same check here

Comment (vlad20012) [reply src/main/kotlin/org/rust/lang/core/resolve/Processors.kt]: Fixed

**Pull request merged**